mechanics/tools/gen-aliases/main.go

```diff
@@ -63,6 +63,11 @@ func main() {
 
 `, infile)
 	output = append([]byte(preamble), output...)
+	prevOut, err := ioutil.ReadFile(outfile)
+	if err == nil && string(prevOut) == string(output) {
+		log.Print("No changes to ", outfile)
+		return
+	}
```

Comment:
Should not write to `outfile` if the content is the same. ie: put `else` for the write.

Reply:
also, to make it clear I would put "No changes needed for " instead of the "No changes to " which might be ambiguous

Reply:
you can remove the `return` when you use `else

Reply:
Made requested changes ✨